### DISCUSSION TITLE
Iniciante no Front-End

### DISCUSSION THREAD

**Discussion by marcelafreire** [frontendbr/forum#1518] [Discussão]
Oi pessoal!

Sou designer, e sempre amei código, porém acabei seguindo mais pelo caminho do design. Agora, gostaria de investir no que realmente gosto, que é o front-end. Porém, só conheço o básico do HMLT e CSS e sei que tenho muito o que aprender. Minha dúvida é: Será que consigo ingressar no mercado front-end? Tenho medo de sair do meu emprego atual para me dedicar ao fornt-end e não conseguir nada na área. Também gostaria de saber em quanto tempo vocês acham que eu consigo aprender o suficiente para ingressar na área?

**Comment by will3g** [comment]
Se você gosta disso, então estude. Está tendo bastante vagas, tanto de estágio quanto CLT e etc... 
Recomendo estudar o básico de Java Script pelo site da Rocketseat (https://rocketseat.com.br/starter). A Alura também têm a formação front-end que pode te ajudar bastante (com bônus como CSS grid e etc...). O tempo depende de quanto você se dedicar aos estudos... Acho que no máximo três meses você já tenha aprendido bastante.

**Comment by Inovassist-dev** [comment]
Pelo que vejo a área fica meio que entre designer-front e programador-front. Você pode aprender js puro, algumas coisas que auxiliam como sass,  e tentar ingressar na área como front-end mais focado em design/ux. Assim seu ingresso no mercado vai ser mais rápido.

**Comment by gersoncjlima** [comment]
Vc já tem o melhor dos dois mundos rsrs, para um próximo degrau é apenas um passo, como os brows comentaram acima, dedicar-se bastante e já, já vc estará num outro patamar. Success!!

**Comment by glauberm** [comment]
Te recomendaria dar uma olhada em sites como o [Codrops](https://tympanus.net/codrops/) e o [Codepen](https://codepen.io/popular/pens/) que têm muito conteúdo que orbitam entre o design e o código.

**Comment by theandersonn** [comment]
@marcelalala Sim! 
As Possibilidades são reais e totalmente alcançáveis, depende apenas de você.

Hoje os processos de Front-End seguem rotinas com tarefas automatizadas, build... São excelentes, mas trazem complexidade que podem confundir quem está iniciando.

Foque inicialmente na raiz, HTML, CSS e JavaScript.

Fontes de estudo que recomendo:
https://www.udemy.com/
https://www.origamid.com/
https://www.schoolofnet.com/
https://www.alura.com.br/ 

Quanto ao tempo, depende de quanto você estudar, colocar em pratica. Creio que um período para maturar as informações é entre 06 meses a 2 anos. Você estará realizando tarefas com confiança.

Profissionais que conseguem reunir skills de designer e front são altamente requisitados no mercado de trabalho.

**Comment by math-jam** [comment]
@marcelalala se você estudar e ter foco, não terá problemas em achar um emprego bom e tal. O pessoal ai em cima passou uma lista de sites de cursos e tal, se você fazer metade do que falaram você já vai estar mais na vantagem que muitos Jr.

**Comment by vitorbritto** [comment]
Interessante sua história pois me identifico em alguns pontos. 

Minha primeira formação foi Analises de Sistema, mas sentia que faltava algo. Faltavam algumas peças do quebra-cabeça. Eu precisava melhorar minha visão desde a concepção até a entrega de um produto que fosse válido para mercado.

Com base nessa inquietude eu ingressei na faculdade de Design. Tive a experiência de viver um mundo que me abriu os olhos e posso dizer com total certeza de que levei alguns tapas na cara até perceber que meu código era Design. Afinal, toda linha de código que faço precisa ser elegante, bem feita, manutenível e replicável.

IMHO, você tem a base mais importante, que é: embasamento e estudo para criar um produto.

Após a faculdade de Design, passei por alguns alguns momentos críticos por conta das escolhas que precisei fazer. Sempre gostei muito de códigos e passei a adorar o Design.

Não me arrependo das escolhas, apesar dos "sofrimentos". Porém, existem determinados momentos em que precisamos escolher. Se não fosse a faculdade de Design, minha visão como Programador seria limitada, eu não saberia o quão importante é usabilidade (louvado seja Jackob Nielsen), navegação amigável, arquitetura de informação, a importância de um planejamento e como não exigir do usuário que ele pense ao tomar decisões ("Não me faça pensar - Steve Krug"), além de coloca no centro de tudo (DCU).

Acredite no seu potencial. Suas escolhas e resultados dependem apenas de você.

Pratique, leia e pratique mais um pouco. Não se afobe, siga seu ritmo. Quando você se sentir confortável, faça algumas entrevistas. Colha os resultados do que você plantou e passe para o próximo nível que é se dedicar exclusivamente ao mundo Front-end.

Minha recomendação é ler o [MDN](https://developer.mozilla.org/pt-BR/docs/Web) e o [Web Fundamentals](https://developers.google.com/web/fundamentals/). 

Como sugestão, leia e deixe o [DevTools do Chrome](https://developers.google.com/web/tools/chrome-devtools/?hl=pt-br) aberto para praticar.


Boa sorte! =]

**Comment by deppbrazil** [comment]
@marcelalala

## Uma ótima maneira de ir ganhando confiança e praticar é:

- Já que vc é Designer desenhe seus layouts e code isso usando apenas HTML5 e CSS3 se tiver a necessidade coloque Javascript, mas sem usar algum frameworks ou lib no inicio, com isso vai conhecendo como desenhar algo nativamente na tela.

- Depois avance, coloque SASS ou outro pré-processador para otimizar seu CSS e verá agilidade e valor em pré-processar e minificar o CSS.

- Em seguida conheça as tags de HTML5 e veja a importância disso em otimização, SEO, acessibilidade e código semântico na web. Boas referencias [MDN](https://developer.mozilla.org/pt-BR/), [W3C](https://www.w3.org/), [W3Schools](https://www.w3schools.com/).

- Depois reproduza grande sites ou sites que vc gosta usando as stacks que vc sabe fazendo um re-brand disso.

- Estude tendências em interface. Boa referência [UpLabs](https://www.uplabs.com/)

- É também estude automatização de tarefas no Frontend. Ferramentas mais comuns para isso [Gulp](https://gulpjs.com/), [Webpack](https://webpack.js.org/).

- Brinque com os frameworks e libs de UI do mercado desde o mais tradicional [Bootstrap](https://getbootstrap.com/) até os mais hype, um exemplo [React](https://reactjs.org/), para ver a diferença e qual problema eles vieram para resolver.

- Seja ativo(a) nas comunidades e acompanhe as issues aqui no FrontendBR e contribua. É serio essa comunidade ajuda demais!

- Procure uma mentoria e se espelhe em alguém que admira dentro do mundo Front-end e veja o que eles contribuem e escrevem nas comunidades e Twitter, 
> Tem muitos caras que escrevem conteúdo ótimos mas minhas referências e preferidos são esses caras, que são foda pra c@r4!H0:
[Felipe Fialho](https://www.felipefialho.com/)
[Raphael Fabeni](https://raphaelfabeni.com/)
[Willan Justen](https://willianjusten.com.br/)
[William Oliveira ](https://woliveiras.com.br/)
[Marco Bruno](https://twitter.com/marcobrunobr)
[Diego Fernandes](https://blog.rocketseat.com.br/author/diego/)
[Túlio Faria](https://www.devpleno.com/author/admin/)
[Fernando Daciuk](https://blog.da2k.com.br/)

- E seria muito bom futuramente ter um blog ou um Medium e escrever sobre coisas que vc aprende no geral.

## Fica ai minha dica ##
# :)
